proposals/expand-ref.md

```diff
@@ -0,0 +1,294 @@
+
+- `__makeref(e)` will be logically treated as a method with the signature `static TypedReference __makeref(ref T value)` were `T` is the type of `e`.
+- `__refvalue(e, T)` 
+    - When `T` is a `ref struct`: will be treated as accessing a field declared as `ref scoped T` inside `e`. 
+    - Will be treated as accessing a field declared as `ref T` inside `e`
+- `__arglist` as a parameter will be implicitly `scoped`
+- `__arglist(...)` as an expression will have a *ref-safe-to-escape* and *safe-to-escape* of *current method*. 
+
+Conforming runtimes will ensure that `TypedReference`, `RuntimeArgumentHandle` and `ArgIterator` are defined as `ref struct`. Further `TypedReference` must be viewed as having a `ref` field to a `ref struct` for any possible type (it can store any value). That combined with the above rules will ensure references to the stack do not escape beyond their lifetime.
+
+Note: strictly speaking this is a compiler implementation detail vs. part of the language. But given the relationship with `ref` fields it is being included in the language proposal for simplicity.
+
+### Annotation Definition
+
+<a name="annotations-param"></a>
+At an annotation level every parameter marked `ref scoped` will have a new lifetime parameter defined. The name will be `$paramN` where _N_ is the numerical order of the parameter. That lifetime will only have the relationship `where $paramN : $local`. 
+
+```csharp
+ref struct S { } 
+void M(ref scoped S s) 
+
+// maps to 
+
+void M<$param1>(ref<$local> S<$param1> s)
+    where $param1 : $cm
+```
+
+This definition prevents the value from escaping from the method as the lifetime is not returnable. It also prevents local data from escaping from the current method through the parameter as the lifetime is wider that `$local` but not equivalent.
+
+```
+void M<$param1>(ref<$local> S<$param1> p)
+    where $param1 : $local
+{
+
+    S<$local> s = new S<$local>(stackalloc int[42]);
+
+    // error: cannot convert S<$local> to S<$param1>
+    p = s;
+}
+```
+
+<a name="annotations-field"></a>
+At an annotation level every field marked `scoped ref` (explicitly or implicitly via `ref scoped`)will have a new lifetime parameter defined. The name will be `$refFieldN` where _N_ is the numerical order of the field. That lifetime will have the relationship `where $refFieldN : $local` in all methods that use the type.
+
+```csharp
+ref struct S
+{
+    scoped ref int i;
+}
+S M(S p) { }
+
+// maps to 
+ref struct S<out $this, $refField1>
+{
+    ref<$refField1> int i;
+}
+
+S<$cm> M<$cm, $l1>M(S<$cm, $l1> p)
+    where $l1 : $local
+{
+
+}
+```
+
+Every field marked as `ref scoped` will have a new lifetime parameter defined. Th name will be `$fieldN` where _N_ is the numerical order of the field. That lifetime will have the relationship `where $fieldN : $refFieldN` defined on the type. It will also have the relationship `where $fieldN : $local` in all method that use the type.
```

Comment:
```suggestion
Every field marked as `ref scoped` will have a new lifetime parameter defined. The name will be `$fieldN` where _N_ is the numerical order of the field. That lifetime will have the relationship `where $fieldN : $refFieldN` defined on the type. It will also have the relationship `where $fieldN : $local` in all method that use the type.
```